### ISSUE TITLE
Import fails when a torrent consists of a single file

### ISSUE THREAD

**Issue by Hunter16g** [karamanolev/WhatManager2#39]
When trying to import a torrent containing example.epub (without any directory), it checks for
/base-dir/example.epub/example.epub
instead of 
/base-dir/example.epub

Creating a directory named "example.epub" and copying example.epub in it solves the problem, but it would be a good thing to correct this if possible.

**Comment by karamanolev** [comment]
Generally, it would be a good thing to fix it, but it is some effort and since these should be replaced by proper torrent-name/file-name.ext torrents (even if they contain a single file), I didn't think it's worth the effort at the time. Maybe just fix them and replace the old ones. If anyone else wants to fix that...

**Comment by Hunter16g** [comment]
Is it mandatory to have a folder even for one file ? I wasn't aware of this rule, in this case it's probably not worth it. karamanolev/WhatManager2 wrote:Generally, it would be a good thing to fix it, but it is some effort and since these should be replaced by proper torrent-name/file-name.ext torrents (even if they contain a single file), I didn't think it's worth the effort at the time. Maybe just fix them and replace the old ones. If anyone else wants to fix that...

—Reply to this email directly or view it on GitHub.


  
  


Sent Using Firefox OS

**Comment by karamanolev** [comment]
Hmmm that seems only to apply to music, but since that's what I use WM for ... I guess anyone who needs that could fix it and I'll merge the pull request.

**Comment by dangerman200k** [comment]
Yeah, I've encountered this too. The rules about single file torrents requiring an enclosing directory only apply to music, so there are valid torrents (books, apps, etc.) that fail to import because of this. I just throw all my torrents in WM rather than having a separate setup for non-music, so it'd be nice to see this fixed. I will see if I can take a look at it myself...
